### DISCUSSION TITLE
allow for current user function which retrieves the user id without hitting the backend strategy

### DISCUSSION THREAD

**Discussion by EricPatel** [fastapi-users/fastapi-users#1430] [Ideas]
I have a use case where I want to use this library to the get current user but I do not want to fetch the user object on every request. This increases the number of DB calls and sometimes just the user id which is stored in the transport is all I need. I have a change to allow the current implementation of  current_user and current_user_token to support this new feature where if you set an optional boolean `id_only` then it will avoid fetching the user object from the user_manager.

**Comment by EricPatel** [comment]
how can I get access to create a pull request with my change to be reviewed?

**Comment by frankie567** [comment]
Good that you achieved what you want 👍 However, I don't think it fits in the common library, since it's a quite specific requirement and can have unintended side effects (e.g. allow a user that's actually deleted in DB). Most authentication frameworks work like this.

But feel free to share your changes in the Discussions forum :)

**Reply by EricPatel** [reply]
I am trying to understand how this could have unintended side effects that couldn't be checked later during runtime. In the example you gave a quick query with a user id against the db could tell me that the user doesn't exist or any rows that have this id as a FK don't exist and I can simply return that. Making a call against Users if we just need the id seems wasteful.

**Reply by frankie567** [reply]
Well, technically yes, but you'll only find out late in the request logic; and it'll probably be hard to tell if it's because the user doesn't exist anymore. Worse, in some cases, you may even be able to fully process the request (if the user itself is not directly needed).

> Making a call against Users if we just need the id seems wasteful.

IMO, that's a micro-optimization. Avoid that request won't really make a difference in terms of performance.

**Reply by EricPatel** [reply]
I would argue the user manager should be able to clear sessions for users that are no longer active and that whole problem goes away. if the user is logged in and their account is deactivate the user manager will go clear the backend session store and now they are no longer auth'd. I consider you adding this feature because again having to make DB reads each time a user makes a request is not scalable and takes away so much bandwidth from the DB.
